hoomd/md/pytest/test_methods.py

```diff
@@ -216,6 +216,69 @@ def test_npt_attributes():
     assert npt.barostat_dof == (1.0, 2.0, 4.0, 6.0, 8.0, 10.0)
 
 
+def test_nph_attributes():
+    """Test attributes of the NPH integrator before attaching."""
+    all_ = hoomd.filter.All()
+    constant_s = [
+        hoomd.variant.Constant(1.0),
+        hoomd.variant.Constant(2.0),
+        hoomd.variant.Constant(3.0),
+        hoomd.variant.Constant(0.125),
+        hoomd.variant.Constant(.25),
+        hoomd.variant.Constant(.5)
+    ]
+    nph = hoomd.md.methods.NPH(filter=all_,
+                               S=constant_s,
+                               tauS=2.0,
+                               couple='xyz')
+
+    assert nph.filter == all_
+    assert len(nph.S) == 6
+    for i in range(6):
+        assert nph.S[i] == constant_s[i]
+    assert nph.tauS == 2.0
+    assert nph.box_dof == (True, True, True, False, False, False)
+    assert nph.couple == 'xyz'
+    assert not nph.rescale_all
+    assert nph.gamma == 0.0
+
+    type_A = hoomd.filter.Type(['A'])
+    nph.filter = type_A
+    assert nph.filter == type_A
+
+    ramp_s = [
+        hoomd.variant.Ramp(1.0, 4.0, 1000, 10000),
+        hoomd.variant.Ramp(2.0, 4.0, 1000, 10000),
+        hoomd.variant.Ramp(3.0, 4.0, 1000, 10000),
+        hoomd.variant.Ramp(0.125, 4.0, 1000, 10000),
+        hoomd.variant.Ramp(.25, 4.0, 1000, 10000),
+        hoomd.variant.Ramp(.5, 4.0, 1000, 10000)
+    ]
+    nph.S = ramp_s
+    assert len(nph.S) == 6
+    for i in range(6):
+        assert nph.S[i] == ramp_s[i]
```

Comment:
Given Python's default implementations for `__eq__` this is equivalent to an `is` check. since currently `Variants` don't implement a nondefault `__eq__` method. This makes this more confusing. If you don't want to test the call method returns the same thing, then use `is` for now even though I suggested the opposite so future developers won't be confused if this tests fails by the `==` check.